### PR TITLE
Fix metric count consensus

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- `Validator signed blocks` displays 0 when running in multikey mode
  
## Proposed changes
- incremenet counter when key is managed by current node

## Testing procedure
- system test + check termui `Validator signed blocks` increases after this fix for the managed keys

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: consensus/spos/bls/subroundStartRound.go [hunk 1]
```diff
@@ -196,8 +196,8 @@ func (sr *subroundStartRound) initCurrentRound() bool {
 		}
 		sr.AppStatusHandler().SetStringValue(common.MetricConsensusState, "not in consensus group")
 	} else {
-		isLeader := leader == sr.SelfPubKey() && sr.ShouldConsiderSelfKeyInConsensus()
-		if !isLeader && !sr.IsKeyManagedByCurrentNode([]byte(leader)) {
+		isLeader := leader == sr.SelfPubKey()
```
Comment (iulianpascalau) [inline consensus/spos/bls/subroundStartRound.go:199]:
```
isLeader := leader == sr.SelfPubKey() || sr.IsKeyManagedByCurrentNode([]byte(leader))
```

File: consensus/spos/bls/subroundStartRound.go [hunk 2]
```diff
@@ -196,8 +196,8 @@ func (sr *subroundStartRound) initCurrentRound() bool {
...
+		if !isLeader && sr.ShouldConsiderSelfKeyInConsensus() {
```
Comment (iulianpascalau) [inline consensus/spos/bls/subroundStartRound.go:200]: I guess we should test that the current node had at least one managed key in the consensus group.

File: consensus/spos/bls/subroundStartRound.go (hunk 1 above)
Comment (sstanculeanu) [reply consensus/spos/bls/subroundStartRound.go:199]: right :man_facepalming:

File: consensus/spos/bls/subroundStartRound.go (hunk 2 above)
Comment (sstanculeanu) [reply consensus/spos/bls/subroundStartRound.go:200]: updated

**Review by miiu96 — APPROVED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 

Normal allin test: v1.5.15-dev-config-e134e5314d -> fix_MetricCountConsensus-60b09789a1

--- Specific errors ---

block hash does not match 4592
wrong nonce in block 1986
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  1
Nr. of all WARNS:  366
Nr. of new ERRORS:  1
Nr. of new WARNS:  3
Nr. of PANICS:  0

/------/

--- ERRORS ---

upcloud-lon-observer-2 :
Error: error calling SaveAccounts, will retry  driver = *host.hostDriver retrial in = error = connection not open while sending data on route for topic SaveAccounts  1

/------/

--- WARNINGS ---

/------/

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**